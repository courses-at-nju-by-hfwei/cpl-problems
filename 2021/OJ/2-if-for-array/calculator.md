# 简易计算器 (calculator.c)

## 题目描述

请实现一个只支持两个**整数**运算的简易计算器，例如：

* 加法：1 + 1 输出 2
* 减法：1 - 2 输出 -1
* 乘法：1 * 9 输出 9
* 除法：9 / 2 输出 4
* 浮点型除法：9 /* 2 输出 4.500（保留 3 位小数）
* 幂：2 ** 4 输出 16
* 模：8 % 5 输出 3

## 输入格式

输入包含多行，每一行的格式为 $a\ {\rm op}\ b$

整数和运算符之间都有空格隔开，保证所有计算结果在 `int` 范围内。

> 教材P41：如果操作数中有负数，C99 规定除法的结果总是趋零结尾的，$i$ % $j$ 的值的符号与 $i$ 的符号相同

数据约定

| op   | $\qquad a, b$ 的范围    |
| ------ | ------------------------- |
| $+$  | $\qquad -10^8\sim 10^8$ |
| $-$  | $\qquad -10^8\sim 10^8$ |
| $*$  | $\qquad -10^4\sim 10^4$ |
| $/$  | $\qquad 1\sim 10^8$     |
| $/*$ | $\qquad 1\sim 10^8$     |
| $**$ | $\qquad 1\sim 9$        |
| %    | $\qquad 1\sim 10^8$     |

## 输出格式

对于每一行输入，输出一行计算结果

## 脚注

**如何读入不定行数据？**

例如读入不定行个整数，可以

```cpp
while (scanf("%d", &n) != EOF) 
{
    // codes...
}
```

在本地测试时可以用 Ctrl + D 结束程序

**你有没有试过用 %d 输出 double 类型的变量呢？输出来的值对吗？math.h 中的 pow() 函数的返回值是什么类型的？**

## 样例

### 样例输入

1 + 1
1 - 2
1 *9
9 / 2
9 /* 2
2 ** 4
8 % 5

### 样例输出

2
-1
9
4
4.500
16
3
