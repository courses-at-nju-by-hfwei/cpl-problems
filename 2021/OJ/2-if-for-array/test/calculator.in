1 + 1
1 - 2
1 * 9
9 / 2
9 /* 2
2 ** 4
8 % 5