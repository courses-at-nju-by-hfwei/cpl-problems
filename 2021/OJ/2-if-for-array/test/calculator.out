2
-1
9
4
4.500
16
3